From 8be0afa4b1c6ce2bebf4727592dd60729843c4a1 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 10 Feb 2020 15:41:12 +0200
Subject: [PATCH] [SW-0] pll: introduce weak "is configurable" api

Introduce the weak variant of the hl_pll_is_configurable() function, so
there will be no need to implement it for each H/W variant (e.g. PLDM,
ONIX, etc.).
Remove the anymore unneeded implementations.
Note, the new weak implementation uses the positive logic (e.g. returns
true) while those removed stubs were using the negative logic (e.g.
return false). This is not a problem as those stubs were never called
because their hl_pll_get_cfg() stub which always returned 0 as the
nummber of PLLs.

Change-Id: I645f4ccaf9f7d5370c5a19bd17f883f77f07d982
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.c b/common/pll.c
index 6a298eb..b4006c1 100644
--- a/common/pll.c
+++ b/common/pll.c
@@ -214,6 +214,9 @@
 	return out_freq;
 }
 
+__attribute__((weak))
+bool hl_pll_is_configurable(struct hl_pll_cfg *cfg) { return true; }
+
 /**
  * hl_pll_init - Initialize PLL registers
  */
diff --git a/gaudi/onix.c b/gaudi/onix.c
index f3e924b..cd7e74e 100644
--- a/gaudi/onix.c
+++ b/gaudi/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/gaudi/pldm.c b/gaudi/pldm.c
index b2a204d..35018b4 100644
--- a/gaudi/pldm.c
+++ b/gaudi/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/goya/onix.c b/goya/onix.c
index 5cc536b..c53ea29 100644
--- a/goya/onix.c
+++ b/goya/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	50000000ULL
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/goya2/onix.c b/goya2/onix.c
index f3e924b..cd7e74e 100644
--- a/goya2/onix.c
+++ b/goya2/onix.c
@@ -13,11 +13,6 @@
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/goya2/pldm.c b/goya2/pldm.c
index b2a204d..35018b4 100644
--- a/goya2/pldm.c
+++ b/goya2/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/goya2_arc/onix.c b/goya2_arc/onix.c
index e922152..8ad210f 100644
--- a/goya2_arc/onix.c
+++ b/goya2_arc/onix.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	57600
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
diff --git a/goya2_arc/pldm.c b/goya2_arc/pldm.c
index d33c141..3f6b3d7 100644
--- a/goya2_arc/pldm.c
+++ b/goya2_arc/pldm.c
@@ -15,11 +15,6 @@
 #define UART_CLOCK_FREQUENCY	(SYS_CLOCK_FREQUENCY / 5)
 #define PLDM_UART_BAUDRATE	1200
 
-bool hl_pll_is_configurable(struct hl_pll_cfg *cfg)
-{
-	return false;
-}
-
 int hl_pll_get_cfg(struct hl_pll_cfg **hl_pll_cfg)
 {
 	return 0;
